applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -391,12 +502,22 @@ void DistanceModificationProcess::RecoverOriginalDistance() {
     this->SetContinuousDistanceToSplitFlag();
 }
 
-void DistanceModificationProcess::RecoverOriginalDiscontinuousDistance() {
+void DistanceModificationProcess::RecoverOriginalDiscontinuousDistance()
+{    
+    // Recover original elemental distances
     #pragma omp parallel for
     for (int i_elem = 0; i_elem < static_cast<int>(mModifiedDistancesIDs.size()); ++i_elem) {
-        const unsigned int elem_id = mModifiedDistancesIDs[i_elem];
-        const auto elem_dist = mModifiedElementalDistancesValues[i_elem];
-        mrModelPart.GetElement(elem_id).SetValue(ELEMENTAL_DISTANCES,elem_dist);
+        const size_t elem_id = mModifiedDistancesIDs[i_elem];
+        const auto r_elem_dist = mModifiedElementalDistancesValues[i_elem];
+        mrModelPart.GetElement(elem_id).SetValue(ELEMENTAL_DISTANCES,r_elem_dist);
+    }
+
+    // Recover original extrapolated elemental edge distances
+    #pragma omp parallel for
```

Comment:
Please use parallel utils.

Reply:
OK, I never did. You mean to replace all `omp parallel for` with either `BlockPartition` or `IndexPartition`?
What about `omp atomic` and `omp critical`?